garage/np/algos/base.py

```diff
@@ -1,7 +1,26 @@
-class Algorithm:
-    pass
+from abc import ABC, abstractmethod
 
 
-class RLAlgorithm(Algorithm):
-    def train(self):
-        raise NotImplementedError
+class RLAlgorithm(ABC):
+    """Base class for all the algorithms.
+    """
+
+    @abstractmethod
+    def process_samples(self, itr, paths):
+        """Return processed sample data based on the collected paths.
+
+        Args:
+            itr (int): Iteration number.
+            paths (list[dict]): A list of collected paths.
+        """
+        pass
+
+    @abstractmethod
+    def train_once(self, itr, paths):
+        """Training once.
```

Comment:
please leave a much more informative description of this interface. if you are unclear on its purpose, ask Keren for an explanation.